scripts/ledger_network_update.py

```diff
@@ -0,0 +1,398 @@
+    spec = importlib.util.spec_from_file_location(module_name, file_name)
+    module = importlib.util.module_from_spec(spec)  # type: ignore
+    sys.modules[module_name] = module
+    spec.loader.exec_module(module)  # type: ignore
+    return module
+
+
+ROOT_DIR = Path(__file__).parent / "../"
+AEA_LEDGER_MODULE_FILE = Path(__file__).parent / "../aea/crypto/ledger_apis.py"
+
+
+class NetworkConfig:
+    """Ledger network configuration data class."""
+
+    net_name: str
+    chain_id: str
+    denom: str
+    rest_api_address: str
+    rpc_api_address: str
+
+    def __init__(
+        self,
+        net_name: str,
+        chain_id: str,
+        denom: str,
+        rest_api_address: str,
+        rpc_api_address: str,
+    ):
+        """
+        Set network config.
+
+        :param net_name: str
+        :param chain_id: str
+        :param denom: str
+        :param rest_api_address: str
+        :param rpc_api_address: str
+        """
+        self.net_name = net_name
+        self.chain_id = chain_id
+        self.denom = denom
+        self.rest_api_address = rest_api_address
+        self.rpc_api_address = rpc_api_address
+
+    def __str__(self) -> str:
+        """Return lines of network configration to be printed."""
+        return (
+            f"Net name: {self.net_name}\n"
+            f"Chain id: {self.chain_id}\n"
+            f"Denom: {self.denom}\n"
+            f"resp api address: {self.rest_api_address}\n"
+            f"rpc api address: {self.rpc_api_address}\n"
+        )
+
+
+class NetworkUpdate:
+    """Ledger network update tool."""
+
+    cur_config: NetworkConfig
+    new_config: NetworkConfig
+    cosmpy_version: str
+
+    @staticmethod
+    def get_current_config() -> NetworkConfig:
+        """
+        Get current ledger network configuration.
+
+        :return: NetworkConfig instance
+        """
+        aea_ledger = load_module(AEA_LEDGER_MODULE_FILE, "aea.crypto.ledger_apis")
+        rest_api_addr = aea_ledger.FETCHAI_DEFAULT_ADDRESS
+        chain_id = aea_ledger.FETCHAI_DEFAULT_CHAIN_ID
+        denom = aea_ledger.FETCHAI_DEFAULT_CURRENCY_DENOM
+        m = re.match(r"https://rest-(\w+).fetch.ai:443", rest_api_addr)
+        if not m:
+            raise ValueError(
+                f"can not determine network name from address: {rest_api_addr}"
+            )
+        net_name = m.groups()[0]
+
+        rpc_api_addr = f"https://rpc-{net_name}.fetch.ai:443"
+
+        return NetworkConfig(
+            net_name=net_name,
+            chain_id=chain_id,
+            denom=denom,
+            rest_api_address=rest_api_addr,
+            rpc_api_address=rpc_api_addr,
+        )
+
+    def get_new_config(self) -> NetworkConfig:
+        """
+        Get new ledger network configuration.
+
+        :return: NetworkConfig instance
+        """
+        net_name = click.prompt("Enter new net name", default=self.cur_config.net_name)
+        chain_id = f"{net_name}-1"
```

Comment:
This is not necessarily -1, sometimes another number. Need to get the whole thing via prompt.

Reply:
it's a default value, that will be proposed during the prompt (a line below)